### PR TITLE
Fixed #30505 -- Prevented migrations detecting changes on field choices order

### PR DESCRIPTION
**Pull request by caioariede**
Trac ticket: https://code.djangoproject.com/ticket/30505

### CODE REVIEW

**Comment by charettes** [comment]
I don't want to be discouraging efforts towards this but it does feel strange that we decide to ignore ordering changes for `choices`  given it has display considerations given we don't do the same for `verbose_name` and `help_text`.

Not that I completely agree with the design decision of the migration framework to track all model fields options even if they don't have database level side effect but I just thought I'd mention this is a step in the opposite direction.

**Comment by caioariede** [comment]
> I don't want to be discouraging efforts towards this but it does feel strange that we decide to ignore ordering changes for choices given it has display considerations and we don't do the same for verbose_name and help_text.

The way the migration handles this is a bit generic (I'm referring to `deep_deconstruction`) that basically calls `deconstruct()` on every field.

If `deconstruct()` is only supposed to work for migrations and not to display choice values or anything else, I think we should be safe.

File: docs/releases/2.2.2.txt [hunk 1]
```diff
@@ -18,3 +18,6 @@ Bugfixes
 * Fixed a regression in Django 2.2.1 where
   :class:`~django.contrib.postgres.search.SearchVector` generates SQL with a
   redundant ``Coalesce`` call (:ticket:`30488`).
+
+* Prevented migrations from detecting changes on field choices when only the
+  ordering has changed (:ticket:`30505`).
```
Comment (charettes) [inline docs/releases/2.2.2.txt]: That's not a regression.

**Comment by felixxm** [comment]
I agree with Simon, I think that Carlton accepted this as a something that should be clarified in a docs. Changing bahavior was not our intention (see [comment](https://code.djangoproject.com/ticket/30505#comment:1)).

**Pull request closed without merging**